programs/utilities/flower.sc

```diff
@@ -1,140 +1,299 @@
+  );
 
-  // gets simplex noise for given  coordinate with seed 2345 (from vanilla code)
-  val = simplex(x / 24, z / 24, null, 2345);
   // makes sure output is between 0 and 0.9999
   val_clamped = min(max(val, 0), 0.9999);
-  flower_id = floor(val_clamped * length(global_FLOWERS));
-);
-
-get_flower_at(position) -> (
-  flower_id = _get_flower_id_at(position);
-  global_FLOWERS : flower_id;
+  flower_id = floor(val_clamped * global_flower_count);
+  // return flower_id
 );
 
 // check for flower forest biome
 _is_flower_forest(position) -> (
   biome(position) == 'flower_forest';
 );
 
-// fills an area with flowers
-__flower_forest_fill(position) -> (
-  flower_pos = pos(position);
-
+// finds and places flower at given pos
+__place_flower_at(position, filter_flower_id) -> (
   // check for flower forest biome
-  if(_is_flower_forest(flower_pos),
+  if(global_setting_ignore_biome || _is_flower_forest(position),
     // gets flower at position
-    found_flower = get_flower_at(flower_pos);
-    __set_flower_at(flower_pos, found_flower);
-    // return 1
-    1,
-  // else
-    // return 0
-    0
-  )
+    found_flower_id = _get_flower_id_at(position);
+
+    // checks if flower is wanted type
+    if(filter_flower_id == null || filter_flower_id == found_flower_id,
+      without_updates(__set_block_at(position, found_flower_id));
+      1, // return 1
+    //else:
+      0; // return 0
+    ),
+  //else:
+    0; // return 0
+  );
 );
 
-// fills an area with flower of given type
-__filtered_flower_forest_fill(position, flower, flower_id) -> (
-  flower_pos = pos(position);
+// places flower/glass at position
+__set_block_at(position, flower_id) -> (
+  if(global_setting_glass_mode,
+    // place glass
+    set(position, global_glass_list : flower_id),
 
-  // check for flower forest biome
-  if(_is_flower_forest(flower_pos),
-    // gets flower at position
-    found_flower_id = _get_flower_id_at(flower_pos);
+  //else:
+    // place grass for flower
+    set(pos_offset(position, 'down'), 'grass_block');
 
-    // checks if flower is wanted type
-    if(found_flower_id == flower_id,
-      __set_flower_at(flower_pos, flower);
-      // return 1
-      1,
-    // else
-      // return 0
-      0
-    ),
-  // else
-    // return 0
-    0
-  )
+    // place flower
+    set(position, global_flower_list : flower_id);
+
+    if(global_setting_place_air_above,
+      // set block above to air (to give space to walk)
+      set(pos_offset(position, 'up'), 'air');
+    );
+  );
 );
 
-// places flower at position
-__set_flower_at(position, flower) -> (
-  // place grass for flower
-  set(pos_offset(position, 'down'), 'grass_block');
+__get_volume_size(pos1, pos2) -> (
+  dx = abs(pos1 : 0 - pos2 : 0) + 1;
+  dy = abs(pos1 : 1 - pos2 : 1) + 1;
+  dz = abs(pos1 : 2 - pos2 : 2) + 1;
 
-  // place flower
-  set(position, flower);
+  dx * dy * dz; // return volume size
+);
+
+__print_error(message) -> (
+  print(player(), format('r ' + message))
+);
```

Comment:
You could just have `exit()` call in here by doing `exit(print(player(), format('r ' + message)))`.